flipt-engine-ffi/src/lib.rs

```diff
@@ -96,6 +97,7 @@ impl Default for EngineOpts {
             update_interval: Some(120),
             reference: None,
             fetch_mode: Some(FetchMode::default()),
+            allow_stale: Some(false),
```

Comment:
how about instead we introduce a new 'error_strategy' option that will allow us to be more flexible in the future (ie we could add retry support at this level too). Example:

```rust
pub enum ErrorStrategy {
    Report,     // (Default) Store the error like we do now
    Fallback,  // Fallback with the previous Snapshot state
}
```

Fallback would be akin to `allow_stale`